frontend/src/app/features/training-and-qualifications/new-training-qualifications-record/new-training/new-training.component.spec.ts

```diff
@@ -273,6 +261,36 @@ describe('NewTrainingComponent', async () => {
 
       expect(noMandatoryTrainingLink).toBeFalsy();
     });
+
+    it('should display a no mandatory training for job role message when mandatory training is not required for the job role', async () => {
+      const { component, fixture } = await setup();
+      component.trainingCategories = [];
+      component.isMandatoryTraining = true;
+      component.workplaceUid = '123';
+      component.missingMandatoryTraining = false;
+      fixture.detectChanges();
```

Comment:
```suggestion
      component.ngOnChanges();
      fixture.detectChanges();
```
For now, this test and the test below seems not passing
Probably we can call ngOnChanges() here to make them pass, because manually editing the variables of the component doesn't seem to trigger the onChange hook

Another way would be to amend the setup() method and add parameters for inputs like `trainingCategories`, `isMandatoryTraining` and `missingMandatoryTraining`. That takes more work but probably reflect the actual data flow better

Reply:
I'd say go for the setup option because it better reflects how the component is used, also I'm not sure why I hadn't thought of passing in an override object for setup functions in tests when I'd suggested the same idea for our mock services, we should definitely start doing this!